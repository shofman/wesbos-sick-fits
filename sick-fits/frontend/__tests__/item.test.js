import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import ItemComponent from '../components/Item'
import Item from '../components/styles/ItemStyles'

const fakeItem = {
  id: '123asdf',
  title: 'A Cool item',
  price: 5000,
  description: 'this item is cool',
  image: 'dog.jpg',
  largeImage: 'largeDog',
}

describe('<Item />', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
  // it('renders and displays properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />)

  //   const PriceTag = wrapper.find('PriceTag')
  //   expect(PriceTag.children().text()).toBe('$50')

  //   expect(wrapper.find('Title a').text()).toBe(fakeItem.title)

  //   const img = wrapper.find('img')
  //   expect(img.props().src).toBe(fakeItem.image)
  //   expect(img.props().alt).toBe(fakeItem.title)
  // })

  // it('renders out the button', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />)

  //   const buttonList = wrapper.find('.buttonList')
  //   expect(buttonList.children()).toHaveLength(3)

  //   expect(buttonList.find('Link').exists()).toBe(true)
  // })
})