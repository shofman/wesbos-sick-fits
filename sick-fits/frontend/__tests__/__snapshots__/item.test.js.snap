// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A Cool item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "123asdf",
          },
        }
      }
    >
      <a>
        A Cool item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    this item is cool
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "123asdf",
          },
        }
      }
    >
      <a>
        Edit 
      </a>
    </Link>
    <AddToCart
      id="123asdf"
    />
    <DeleteItem
      id="123asdf"
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
